### PR TITLE
adv tasking: protected entries and guard before quiz using them

### PR DESCRIPTION
**Pull request by leogermond**

### CODE REVIEW

File: courses/fundamentals_of_ada/adv_240_tasking.rst [hunk 1]
```diff
@@ -625,6 +625,61 @@ Waiting On Different Entries
...
 
+------------------------
+Protected Object Entries
+------------------------
+
+* **Special** kind of protected :ada:`procedure`
+* May use a **barrier**, that **only** allows call on a **boolean** condition
+* Barrier is **evaluated** and may be **relieved** when
+
+   - A task calls :ada:`entry`
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_240_tasking.rst:636]: I'm not sure what "A task calls 'entry'" means.
I think you really mean is "A task calls the entry" (i.e. not the keyword, but the concept)
(I think the next line is OK - re-evaluate the guard whenever a protected object is modified)

File: courses/fundamentals_of_ada/adv_240_tasking.rst [hunk 2]
```diff
@@ -625,6 +625,61 @@ Waiting On Different Entries
...
+
+   protected body Stack is
+      entry Push (V : Integer) when Size < Buffer'Length is
+      ...
+      entry Pop  (V : out Integer) when Size > 0 is
+      ...
+   end Object;
+
+-------------------
+Guard Expressions
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_240_tasking.rst:653]: Ok, more confusion :(
A Guard Expression is exactly what we just called a "barrier" in the previous slide.
I would move this slide to before "Protected Object Entries", and then remove the reference to protected objects on this slide (as we'll discuss it in the following slide)

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_240_tasking.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_240_tasking.rst]: Guard and barrier are actually two different concepts in the ARM, I tried clarifying that, WDYT

Comment (frank-at-adacore) [reply courses/fundamentals_of_ada/adv_240_tasking.rst]: OK, now _I_ understand it better :)

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**